packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -159,9 +186,28 @@ const ComboBoxButton = React.forwardRef(function ComboBoxButton(props: ComboBoxB
       valueId,
       validationState === 'invalid' ? invalidId : null
     ].filter(Boolean).join(' '),
-    elementType: 'div'
+    elementType: 'div',
+    // Prevent focus on press and focus on press end to avoid situations where user
+    // presses and holds on the combobox button and tray immediately opens due to menuTrigger=focus
```

Comment:
```suggestion
    // Prevent focus on `press` and instead focus on `pressend` to avoid situations where users
    // press and holds the combobox button and tray immediately opens due to menuTrigger=focus
```